syncer/syncer.go

```diff
@@ -1358,6 +1351,13 @@ func (s *Syncer) syncDML(
 
 			if idx >= count || sqlJob.tp == flush {
 				affect, err = executeSQLs()
+
+				failpoint.Inject("SafeModeExit", func(val failpoint.Value) {
```

Comment:
Shall we put this failpoint into `executeSQLs`

Reply:
take a look at the above comment in "conversation" tab

and I'll add the length check of `job` to fix panic soon

Reply:
done